packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -8,11 +8,29 @@ import { SupportError } from './errors/index.mjs'
  *
  * @internal
  * @abstract
+ * @template {Element} [ModuleType=HTMLElement]
  */
 export class GOVUKFrontendComponent {
+  /**
+   * Defines the type of `Element` the components expects at instantiation
+   *
+   * If the component requires a more specialised type:
+   * - document the child class using `@extends`,
+   *   setting the required type in the generic part of `GOVUKFrontendComponent`.
```

Comment:
Can we be explicit about using the `@augments` annotation to do this?

I'm also thinking it'd be good to think about what's in our contributing docs before v5. For example, I can see it being helpful having a "creating a new component" doc which takes a user step by step through creating a component and includes tips on specific stuff like this.

Or maybe a more generic "working with types" doc, since I'm guessing a lot of potential contributors are not going to be super familiar with them (we could also document "how to turn off type checking" to streamline the process for contributors, then take the actual typing work on ourselves, so we don't block folks).

Not necessarily something to do now, but maybe worth getting around to. @claireashworth - dunno if the [contributing docs](https://github.com/alphagov/govuk-frontend/tree/main/docs/contributing) are on the radar at all?

Reply:
Do you mean providing a clear `@extends GOVUKFrontendComponent<HTMLAnchorElement>` example?

Agreed on the need for some docs describing how we use TypeScript, links to interesting places and maybe recurring patterns in our code.

Reply:
Yeah, it's `@augments GOVUKFrontendComponent<HTMLAnchorElement>` on the Skip Link, but good to have it here to show how its done without having to go look there.

Reply:
Ohh, I don't remember typing `@augments`, maybe something linted it that way.